### PR TITLE
[Windows] remote control re-design.

### PR DESCRIPTION
**Pull request by afedchin**
see commits

### CODE REVIEW

File: xbmc/windowing/win10/WinEventsWin10.cpp [hunk 1]
```diff
@@ -182,6 +185,11 @@ void CWinEventsWin10::InitEventHandlers(CoreWindow^ window)
     m_smtc->IsEnabled = true;
     CAnnouncementManager::GetInstance().AddAnnouncer(this);
   }
+  if (CSysInfo::GetWindowsDeviceFamily() == CSysInfo::WindowsDeviceFamily::Xbox)
+  {
+    m_remote = std::make_unique<CRemoteControlXbox>();
```
Comment (garbear) [inline xbmc/windowing/win10/WinEventsWin10.cpp]: `std::make_unique` is c++14, use `m_remote.reset(new CRemoteControlXbox);`

**Review by FernetMenta — APPROVED** [review]
great

File: xbmc/windowing/win10/WinEventsWin10.cpp (hunk 1 above)
Comment (afedchin) [reply xbmc/windowing/win10/WinEventsWin10.cpp]: this is windows code only. msvc supports some c++14 features.

**Review by garbear — APPROVED** [review]
Good reduction of our maintenance burden

File: xbmc/platform/win32/input/IRServerSuite.cpp [hunk 2]
```diff
@@ -83,41 +66,41 @@ void CRemoteControl::Close()
...
 
-void CRemoteControl::Process()
+void CIRServerSuite::Process()
 {
-  unsigned int iMsRetryDelay = 5000;
-  int iAttempt = 0;
+  m_profileId = CServiceBroker::GetProfileManager().GetCurrentProfileId();
+  m_irTranslator.Load(IRSS_MAP_FILENAME);
+
   // try to connect 60 times @ a 5 second interval (5 minutes)
```
Comment (Rechi) [inline xbmc/platform/win32/input/IRServerSuite.cpp]: comment is now wrong as the retry limiter was removed

**Pull request merged**